csrc/kernel_ir.cpp

```diff
@@ -571,7 +571,11 @@ Val* ForLoop::step() const {
 }
 
 Val* ForLoop::simplifiedStop() const {
-  return simplifyExpr(stop());
+  if (simplified_stop_ == nullptr) {
+    simplified_stop_ =
+        GpuLower::current()->commonScalarMap().hoistScalar(stop(), {});
+  }
+  return simplified_stop_;
 }
```

Comment:
Without this change, I am seeing a string compare failure in `ARangeScalarHoisting1`, whose expect is

```CUDA
__global__ void CUDAGeneratedKernel(int64_t i0, int64_t i1, int64_t i2, Tensor<int64_t, 1, 1> T0, Tensor<int64_t, 1, 1> T1) {
  int64_t i3;
  i3 = i1 - i0;
  int64_t i4;
  i4 = abs(i3);
  int64_t i5;
  i5 = abs(i2);
  int64_t i6;
  i6 = ceilDiv(i4, i5);
  #pragma unroll 1
  for(nvfuser_index_t i7 = 0; i7 < i6; ++i7) {
    T0[i7] = (i0 + (i2 * i7));
  }
  #pragma unroll 1
  for(nvfuser_index_t i8 = 0; i8 < i6; ++i8) {
    T1[i8] = i6;
  }
}
```
actually got
```CUDA
__global__ void CUDAGeneratedKernel(int64_t i0, int64_t i1, int64_t i2, Tensor<int64_t, 1, 1> T0, Tensor<int64_t, 1, 1> T1) {
  int64_t i3;
  i3 = i1 - i0;
  int64_t i4;
  i4 = abs(i3);
  int64_t i5;
  i5 = abs(i2);
  int64_t i6;
  i6 = ceilDiv(i4, i5);
  #pragma unroll 1
  for(nvfuser_index_t i7 = 0; i7 < ceilDiv(abs((i1 - i0)), abs(i2)); ++i7) {
    T0[i7] = (i0 + (i2 * i7));
  }
  #pragma unroll 1
  for(nvfuser_index_t i8 = 0; i8 < ceilDiv(abs((i1 - i0)), abs(i2)); ++i8) {
    T1[i8] = i6;
  }
}
```
This failure is because `simplifyExpr` no longer preserve the `Val*` after the change of this PR, so during codegen, a whole expr is generated. So I decide to make this change to restore back the previous string.